Wordle.Api/Wordle.Api/Services/WordOfTheDayService.cs

```diff
@@ -1,20 +1,51 @@
-﻿namespace Wordle.Api.Services;
+﻿using Microsoft.EntityFrameworkCore;
+using Wordle.Api.Models;
+
+namespace Wordle.Api.Services;
 
 public class WordOfTheDayService
 {
     private readonly List<string> words = LoadWordList();
+    private static object _lock = new();
 
-    private static string? Word = null;
+    public WordleDbContext Db { get; set; }
+
+    public WordOfTheDayService(WordleDbContext db)
+    {
+        Db = db;
+    }
 
     public string GetRandomWord()
     {
-        if (Word == null)
+        Random random = new();
+        int index = random.Next(words.Count);
+        return words[index];
+    }
+
```

Comment:
You will also need a model cs file for the leaderboard implementation. This is where you add player and the relevant variables like PlayerId, Name, GameCount, and AverageAttempts per assignment